importer/SeBbrSv.py

```diff
@@ -49,38 +49,87 @@ def set_heritage_bbr(self):
         querying the source database via their API.
         """
         protection_date = False
-        url = "http://kulturarvsdata.se/" + \
-            self.wd_item["statements"][
-                self.props["cultural_heritage_sweden"]][0]["value"]
+        kulturarv_id = self.wd_item["statements"][
+            self.props["cultural_heritage_sweden"]][0]["value"]
+        url = "http://kulturarvsdata.se/{}".format(kulturarv_id)
+        # http://kulturarvsdata.se/raa/bbr/21300000023251
         url_list = url.split("/")
         url_list.insert(-1, "jsonld")
         url = "/".join(url_list)
-        data = requests.get(url).json()
+        # Request data in json format
+        # http://kulturarvsdata.se/raa/bbr/jsonld/21300000023251
+        data = requests.get(url)
+        try:
+            data = data.json()
+        except ValueError:
+            # Got a jsondecodeerror once, on bbr: 21420000015942
+            # utils.get_bbr_link gives a None from it.
+            # If it occurs, skip the heritage statement,
+            # but the question is whether the item shouldn't be skipped...
+            self.add_to_report("kulturarv_url", url)
+            return
         for element in data["@graph"]:
             if "ns5:spec" in element:
                 bbr_type = element["ns5:spec"]
                 if bbr_type.startswith("Kyrkligt kulturminne"):
+                    # Kyrkligt kulturminne. 4 kap. KML -- these don't seem to have dates
                     type_q = "Q24284073"
                 elif bbr_type.startswith("Byggnadsminne"):
+                    # Byggnadsminne (BM) 3 kap. KML (1977-02-25)
                     type_q = "Q24284072"
                     protection_date = bbr_type.split("(")[-1][:-1]
                 elif bbr_type.startswith("Statligt byggnadsminne"):
+                    # Statligt byggnadsminne (SBM). Förordning (2013:558) (1935-01-25)
                     type_q = "Q24284071"
                     protection_date = bbr_type.split("(")[-1][:-1]
-        """
-        The original set_heritage() added an empty claim
-        because there's no heritage status specified in mapping file,
-        so we start by removing that empty claim.
-        """
+        # The original set_heritage() added an empty claim
+        # because there's no heritage status specified in mapping file,
+        # so we start by removing that empty claim.
         self.remove_statement("heritage_status")
         if protection_date:
             # 1969-01-31
-            date_dict = utils.date_to_dict(protection_date, "%Y-%m-%d")
-            qualifier = {"start_time":
-                         {"time_value": date_dict}}
+            try:
+                date_dict = utils.date_to_dict(protection_date, "%Y-%m-%d")
+                qualifier = {"start_time":
+                             {"time_value": date_dict}}
+            except ValueError:
+                self.add_to_report("protection_type", bbr_type)
```

Comment:
I guess this depends on whether we see a heritage status with a malformed date as valuable. I think if the status looks OK but the date is skipped it can still be added.

I've moved the `qualifier = None` up.